products/dialogflow/api.yaml

```diff
@@ -129,4 +129,129 @@ objects:
+      is an extraction of a user command or sentence semantics.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/dialogflow/docs/'
+      api: 'https://cloud.google.com/dialogflow/docs/reference/rest/v2/projects.agent.intents'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        output: true
+        description: |
+          The unique identifier of this intent. 
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        required: true
+        description: |
+          The name of this intent to be displayed on the console.
+      - !ruby/object:Api::Type::Enum
+        name: 'webhookState'
+        description: |
+          Indicates whether webhooks are enabled for the intent.
+          * WEBHOOK_STATE_UNSPECIFIED: Webhook is disabled in the agent and in the intent.
+          * WEBHOOK_STATE_ENABLED: Webhook is enabled in the agent and in the intent.
+          * WEBHOOK_STATE_ENABLED_FOR_SLOT_FILLING: Webhook is enabled in the agent and in the intent. Also, each slot
+          filling prompt is forwarded to the webhook.
+        values:
+          - :WEBHOOK_STATE_UNSPECIFIED
+          - :WEBHOOK_STATE_ENABLED
+          - :WEBHOOK_STATE_ENABLED_FOR_SLOT_FILLING
+      - !ruby/object:Api::Type::Integer
+        name: 'priority'
+        description: |
+          The priority of this intent. Higher numbers represent higher priorities.
+            - If the supplied value is unspecified or 0, the service translates the value to 500,000, which corresponds
+            to the Normal priority in the console.
+            - If the supplied value is negative, the intent is ignored in runtime detect intent requests.
+      - !ruby/object:Api::Type::Boolean
+        name: 'isFallback'
+        description: |
+         Indicates whether this is a fallback intent.
+      - !ruby/object:Api::Type::Boolean
+        name: 'mlDisabled'
+        description: |
+         Indicates whether Machine Learning is disabled for the intent.
+         Note: If mlDisabled setting is set to true, then this intent is not taken into account during inference in ML
+         ONLY match mode. Also, auto-markup in the UI is turned off.
+      - !ruby/object:Api::Type::Array
+        name: 'inputContextNames'
+        item_type: Api::Type::String
+        description: |
+         The list of context names required for this intent to be triggered.
+         Format: projects/<Project ID>/agent/sessions/-/contexts/<Context ID>.
+      - !ruby/object:Api::Type::Array
+        name: 'events'
+        item_type: Api::Type::String
+        description: |
+         The collection of event names that trigger the intent. If the collection of input contexts is not empty, all of
+         the contexts must be present in the active user session for an event to trigger this intent.
```

Comment:
Is this completely unstructured or are there only certain events that the API will accept? It might help to link to a page that describes what 'events' are in this context.